Mathlib/RingTheory/Polynomial/DegreeLT.lean

```diff
@@ -0,0 +1,191 @@
+  rw [addLinearEquiv, Basis.equiv_apply, Equiv.refl_apply, basisProd_castAdd]
+
+lemma addLinearEquiv_natAdd (i : Fin n) :
+    addLinearEquiv R m n (basis R (m+n) (i.natAdd m)) = (0, basis R n i) := by
+  rw [addLinearEquiv, Basis.equiv_apply, Equiv.refl_apply, basisProd_natAdd]
+
+lemma addLinearEquiv_symm_apply_inl_basis (i : Fin m) :
+    (addLinearEquiv R m n).symm (LinearMap.inl R _ _ (basis R m i)) = basis R (m+n) (i.castAdd n) :=
+  (LinearEquiv.symm_apply_eq _).2 (addLinearEquiv_castAdd i).symm
+
+lemma addLinearEquiv_symm_apply_inr_basis (j : Fin n) :
+    (addLinearEquiv R m n).symm (LinearMap.inr R _ _ (basis R n j)) = basis R (m+n) (j.natAdd m) :=
+  (LinearEquiv.symm_apply_eq _).2 (addLinearEquiv_natAdd j).symm
+
+lemma addLinearEquiv_symm_apply_inl (P : R[X]_m) :
+    ((addLinearEquiv R m n).symm (LinearMap.inl R _ _ P) : R[X]) = (P : R[X]) := by
+  rw [← (basis ..).sum_repr P]
+  simp [-LinearMap.coe_inl, addLinearEquiv_symm_apply_inl_basis]
+
+lemma addLinearEquiv_symm_apply_inr (Q : R[X]_n) :
+    ((addLinearEquiv R m n).symm (LinearMap.inr R _ _ Q) : R[X]) = (Q : R[X]) * X ^ (m : ℕ) := by
+  rw [← (basis ..).sum_repr Q]
+  simp [-LinearMap.coe_inr, Finset.sum_mul, addLinearEquiv_symm_apply_inr_basis,
+    smul_eq_C_mul, mul_assoc, ← pow_add, add_comm]
+
+lemma addLinearEquiv_symm_apply (PQ) :
+    ((addLinearEquiv R m n).symm PQ : R[X]) = (PQ.1 : R[X]) + (PQ.2 : R[X]) * X ^ (m : ℕ) := calc
+  _ = ((addLinearEquiv R m n).symm (LinearMap.inl R _ _ PQ.1 + LinearMap.inr R _ _ PQ.2) : R[X]) :=
+      by rw [LinearMap.inl_apply, LinearMap.inr_apply, Prod.add_def, add_zero, zero_add]
+  _ = _ := by rw [map_add, Submodule.coe_add,
+        addLinearEquiv_symm_apply_inl, addLinearEquiv_symm_apply_inr]
+
+lemma addLinearEquiv_symm_apply' (PQ) :
+    ((addLinearEquiv R m n).symm PQ : R[X]) = (PQ.1 : R[X]) + X ^ (m : ℕ) * (PQ.2 : R[X]) := by
+  rw [X_pow_mul, addLinearEquiv_symm_apply]
+
+lemma addLinearEquiv_apply' {R : Type*} [Ring R] (f) :
+    ((addLinearEquiv R m n f).1 : R[X]) = f %ₘ (X ^ m) ∧
+      ((addLinearEquiv R m n f).2 : R[X]) = f /ₘ (X ^ m) := by
+  rw [and_comm, eq_comm, eq_comm (b := _ %ₘ _)]
+  nontriviality R; refine div_modByMonic_unique _ _ (monic_X_pow _) ⟨?_, ?_⟩
+  · rw [← addLinearEquiv_symm_apply', LinearEquiv.symm_apply_apply]
+  · rw [degree_X_pow, ← mem_degreeLT]; exact Subtype.prop _
+
+lemma addLinearEquiv_apply_fst {R : Type*} [Ring R] (f) :
+    ((addLinearEquiv R m n f).1 : R[X]) = f %ₘ (X ^ m) :=
+  (addLinearEquiv_apply' f).1
+
+lemma addLinearEquiv_apply_snd {R : Type*} [Ring R] (f) :
+    ((addLinearEquiv R m n f).2 : R[X]) = f /ₘ (X ^ m) :=
+  (addLinearEquiv_apply' f).2
+
+lemma addLinearEquiv_apply {R : Type*} [Ring R] (f) :
+    addLinearEquiv R m n f =
+      (⟨f %ₘ (X ^ m), addLinearEquiv_apply_fst f ▸ Subtype.prop _⟩,
+      ⟨f /ₘ (X ^ m), addLinearEquiv_apply_snd f ▸ Subtype.prop _⟩) :=
+  Prod.ext (Subtype.ext <| addLinearEquiv_apply_fst f) (Subtype.ext <| addLinearEquiv_apply_snd f)
```

Comment:
Can't these be proved without appealing to `basis`?

Reply:
I don't think I can.